### PR TITLE
MCP fields projection: evaluate relation-count field access via the real pipeline

### PR DESCRIPTION
**Pull request by borisno2**
## Summary

Small correctness follow-up to #1007 (MCP `query` tool `fields` projection, issue #851), which merged one commit short of this fix due to a race with auto-merge — the PR's own CI had gone green on the second-to-last commit, and auto-merge fired before this last push registered. This applies exactly that dropped commit on top of the now-merged code, nothing else.

**The gap:** #1007 added a check so a to-many relation's `count` couldn't bypass the relationship *field's* own field-level `read` access (`_count` isn't a declared field, so `filterReadableFields` never applied that check to it the way it already does for the relation's own rows). The first version of that fix re-evaluated `checkFieldAccess` directly against the row `context.db` had already returned — but that row had already been through field-visibility once. If the field's access rule depends on some *other* field's raw value that the first pass had already stripped (denied or unselected), the second, independent evaluation could reach a different answer than the canonical one.

**The fix:** a counted relation is now always named in the include (a count-only one with `take: 0`, fetching no rows), so the *ordinary* read pipeline evaluates the relationship field's access exactly once, against the true raw row, in the one place that already does this correctly (`filterReadableFields`). `projectMcpResult` now just reads that decision off whether the relation's key survived into the already-filtered result — no redundant access evaluation — and is synchronous again as a result.

- `packages/core/src/mcp/projection.ts` — count-only relations get a `{ take: 0 }` include entry; `projectMcpResult` drops the `checkFieldAccess` re-evaluation and `async`, checking `key in rawItem` instead.
- `packages/core/src/mcp/handler.ts` — simplified back to a plain `.map()` now that `projectMcpResult` is synchronous.
- Tests: a wire-level assertion that the `take: 0` include entry is built (`mcp-handler.test.ts`), plus an end-to-end test through the real `getContext` pipeline proving a relationship-field-denied count is actually suppressed (`mcp-fields-projection-access.test.ts`).

No behavior change for any case the shipped version already got right — this only changes how the decision is computed, for the edge case where it could previously disagree with the canonical one.

## Test plan

- [x] `pnpm test` (`packages/core`) — 1166/1166 pass.
- [x] `pnpm build` — typechecks clean.
- [x] `pnpm lint` / `pnpm manypkg fix` / `pnpm format` — clean, no new warnings.
- [x] No changeset needed — this refines an unreleased fix from the same in-flight minor bump (#1007's changeset already covers the feature; nothing published between the two).

---
_Generated by [Claude Code](https://claude.ai/code/session_01CthNCkBfwh5G3HGmG1AVLK)_

### CODE REVIEW

**Comment by borisno2** [comment]
Correct — `take: 0` was wrong in exactly the opposite direction from what I was trying to fix: a value-dependent access rule (e.g. `read: ({item}) => item.comments.length === 0`) would see a permanently-empty array under `take: 0` regardless of the relation's true content, potentially GRANTING access it should deny.

Fixed in cd34e3f: a count-only relation now fetches the same real rows a `fields`-and-`count` request would get (default take 5, same as "alongside" mode) — they just never reach the projected output, since `fieldSelection` has no entry for that key. Field-visibility's read-access check now always runs against the true fetched value, whether the caller asked for rows, a count, or both.

Added an end-to-end test through the real pipeline reproducing your exact scenario (`commentsVisibleIfEmpty`, denying read when the relation is non-empty) — it would have failed under the `take: 0` version and passes now. Full suite green (1167/1167).

---
_Generated by [Claude Code](https://claude.ai/code)_

File: packages/core/src/mcp/projection.ts [hunk 1]
```diff
@@ -399,6 +398,16 @@ export async function resolveFieldsProjection(
...
       fieldSelection[fieldName] = { _type: 'fragment', _fields: withId(nestedSelection) }
+    } else if (wantsCount) {
+      // Count-only (no `fields`): still name the relation in the include,
+      // fetching zero rows, purely so the ordinary read pipeline evaluates
+      // the relationship FIELD's own field-level `read` access for it the
+      // same way it would for any other relation — `projectMcpResult` below
+      // then reads that decision off whether the key survived, rather than
+      // re-deriving it itself against a row `filterReadableFields` may have
+      // already stripped other fields from.
+      include[fieldName] = { take: 0 }
```
[the remark this answers is not in the document]
Comment (borisno2) [reply packages/core/src/mcp/projection.ts]: Fixed in cd34e3f — a count-only relation now fetches the same real rows a `fields`-and-`count` request would get, so field-visibility's read-access check always sees the true value. Full details in the PR-level comment. Resolving this thread.

---
_Generated by [Claude Code](https://claude.ai/code)_
